src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/FileBinding.java

```diff
@@ -49,7 +49,9 @@
     }
 
     @Override public SingleEnvironment bindSingle(Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-        FileCredentials credentials = getCredentials(build);
+        if(workspace==null)
```

Comment:
I guess it is a side effect of calling `bind` with null values, which you should not do.